Lib/typing.py

```diff
@@ -2445,23 +2441,23 @@ def final(f):
     method cannot be overridden, and decorated class cannot be subclassed.
     For example:
 
-      class Base:
-          @final
-          def done(self) -> None:
-              ...
-      class Sub(Base):
-          def done(self) -> None:  # Error reported by type checker
+        class Base:
+            @final
+            def done(self) -> None:
+                ...
+        class Sub(Base):
+            def done(self) -> None:  # Error reported by type checker
                 ...
 
-      @final
-      class Leaf:
-          ...
-      class Other(Leaf):  # Error reported by type checker
-          ...
+        @final
+        class Leaf:
+            ...
+        class Other(Leaf):  # Error reported by type checker
+            ...
 
     There is no runtime checking of these properties. The decorator
-    sets the ``__final__`` attribute to ``True`` on the decorated object
-    to allow runtime introspection.
+    attempts to set the ``__final__`` attribute to ``True`` on the decorated
```

Comment:
If `AttributeError` or `TypeError` is encountered, the `__final__` attribute won't be set.